pyrefly/lib/test/class_overrides.rs

```diff
@@ -150,6 +150,28 @@ class Derived2(Base):
     "#,
 );
 
+testcase!(
```

Comment:
The test coverage here seems rather sparse for such a complex feature. Could you also add tests for these?
- a method in a subclass that does override a method in a parent but doesn't use the `@override` decorator
- a method that uses both decorator hints and parent hints
- a static method and a class method
- a method that uses more parameter types than just Pos (PosOnly, Kwarg, etc.)